homeassistant/components/sensor/mqtt.py

```diff
@@ -24,6 +24,8 @@
 from homeassistant.helpers.event import async_track_point_in_utc_time
 from homeassistant.util import dt as dt_util
 
+REQUIREMENTS = ['flatten_json==0.1.6']
```

Comment:
I mean generally in Home Assistant we try to offload as much  as possible into external libraries. But in this case the helper method is quite easy to do on your own. So I don't see why we should introduce a requirement for this simple helper.

Reply:
I don't really agree. This would add around 50 lines to a 180 lines file and moreover is really functionality that does not belong in a mqtt module.
also https://github.com/amirziai/flatten seems to be well supported and test covered.

Reply:
I think I can do it in 19 lines:

```python
def flatten(nested_dict):
    flattened_dict = {}
    
    def _flatten(obj, key):
        if not obj:
            flattened_dict[key] = obj
            return
        base = '' if key is None else key + '.'
        if isinstance(obj, dict):
            for k, v in obj.items():
                _flatten(v, base + str(k))
        elif isinstance(obj, list):
            for index, item in enumerate(obj):
                _flatten(item, base + str(index))
        else:
            flattened_dict[key] = obj

    _flatten(nested_dict, None)
    return flattened_dict
```

You are right that it doesn't belong in the MQTT module, but that doesn't stop us from putting it in a helper. Maybe some other platforms would like to use it too (like the template sensor platform). Ultimately it's not up to me, but I feel like requiring a whole library for one small helper is overkill.